gen/service.go

```diff
@@ -280,3 +286,37 @@ func functionUnwrapResponse(g Generator, f *compile.FunctionSpec) (string, error
 			}
 		`, f)
 }
+
+func functionArgsEnveloper(g Generator, f *compile.FunctionSpec) error {
+	return g.DeclareFromTemplate(
+		`
+		<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+		<$argType := printf "%v%v" (goCase .Name) "Args">
+		<$v := newVar "v">
+
+		func (<$v> *<$argType>) MethodName() string {
+			return "<.MethodName>"
+		}
+
+		func (<$v> *<$argType>) Type() <$wire>.EnvelopeType {
```

Comment:
I meant that `Type()` is okay because `type` is already a reserved keyword, but yeah, `EnvelopeType` works better because `Type` is fairly general.